Sources/WordPressUI/Extensions/UIImageView+Networking.swift

```diff
@@ -131,47 +134,63 @@ public extension UIImageView {
     }
 
     /// Cancels the current download task and clear the downloadURL
-    ///
     @objc func cancelImageDownload() {
-        downloadTask?.cancel()
-        downloadTask = nil
+        context.cancel()
     }
 
     /// Returns a URLRequest for an image, hosted at the specified URL.
-    ///
     private func request(for url: URL) -> URLRequest {
         var request = URLRequest(url: url)
         request.httpShouldHandleCookies = false
         request.addValue("image/*", forHTTPHeaderField: "Accept")
-
         return request
     }
 
-
-    /// Stores the current DataTask, in charge of downloading the remote Image.
-    ///
-    private var downloadTask: URLSessionDataTask? {
-        get {
-            return objc_getAssociatedObject(self, &Downloader.taskKey) as? URLSessionDataTask
-        }
-        set {
-            objc_setAssociatedObject(self, &Downloader.taskKey, newValue as AnyObject, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+    /// Stores image download context.
+    private var context: ImageDownloadContext {
+        if let context = objc_getAssociatedObject(self, &ImageDownloadContext.contextKey) as? ImageDownloadContext {
+            return context
         }
+        let context = ImageDownloadContext()
+        objc_setAssociatedObject(self, &ImageDownloadContext.contextKey, context as AnyObject, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        return context
     }
+}
 
-    /// Private helper structure
-    ///
-    private struct Downloader {
-        /// Key used to associate the current URL.
-        ///
-        static var urlKey = "urlKey"
-
-        /// Key used to associate a Download task to the current instance.
-        ///
-        static var taskKey = "downloadTaskKey"
+// MARK: - Helpers
+
+private final class ImageDownloadContext {
+    var taskId = 0
+    var task: URLSessionDataTask?
+
+    static var contextKey: UInt8 = 0
+
+    func getNextTaskId() -> Int {
+        taskId += 1
+        return taskId
+    }
+
+    func cancel() {
+        taskId += 1
+        task?.cancel()
+        task = nil
     }
 }
 
+private func makeImage(with data: Data) -> UIImage? {
+    guard !data.isEmpty, let image = UIImage(data: data) else {
+        return nil
+    }
+    if #available(iOS 15.0, *) {
+        // Decompress the image in the background
+        return image.preparingForDisplay() ?? image
```

Comment:
Decompressed the images in the background. This is covered by https://developer.apple.com/videos/play/wwdc2021/10252/